### PR TITLE
DHFPROD-3994:CRUD matching artifact

### PR DESCRIPTION
**Pull request by srinathgit**
CRUD matching artifact

### CODE REVIEW

File: one-ui/ui/src/util/roles.tsx [hunk 1]
```diff
@@ -33,6 +35,12 @@ export class RolesService implements IRolesContextInterface {
     public canWriteMappings:() => boolean = () => {
         return this.roles.includes('data-hub-mapping-writer');
     };
+    public canReadMatching:() => boolean = () => {
```
Comment (rjrudin) [inline one-ui/ui/src/util/roles.tsx]: @hao1st Are we at a point where these can become capabilities so we're not exposing role names to the GUI? Or is the plan to take care of that in a separate refactoring ticket?

Comment (hao1st) [reply one-ui/ui/src/util/roles.tsx]: @rjrudin I just created a task to remove role names as login response once the UI does not use roles any more.
https://project.marklogic.com/jira/browse/DHFPROD-4583

File: one-ui/ui/src/components/entities/matching/create-edit-matching-dialog/create-edit-matching-dialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  beforeEach(() => {
+    wrapper = shallow(<CreateEditMatchingDialog data/>)
+  });
+
+  test('Create/Edit Matching Dialog renders ', () => {
+    expect(wrapper.find('#name').length).toEqual(1)
+    expect(wrapper.find('#description').length).toEqual(1)
+    expect(wrapper.find('#srcType').length).toEqual(1)
+    expect(wrapper.find('#srcQuery').length).toEqual(1)
```
Comment (bsrikan) [inline one-ui/ui/src/components/entities/matching/create-edit-matching-dialog/create-edit-matching-dialog.test.tsx]: this should equal 0. Collection radio box is checked by default so srcQuery input div will be hidden

File: one-ui/ui/src/util/roles.tsx (hunk 1 above)
Comment (rjrudin) [reply one-ui/ui/src/util/roles.tsx]: Sounds good

**Comment by srinathgit** [comment]
Had to change the role names "data-hub-matching-reader" and "data-hub-matching-writer" to "data-hub-match-merge-reader" and "data-hub-match-merge-writer"

File: marklogic-data-hub/src/main/resources/hub-internal-config/security/roles/data-hub-match-merge-reader.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "role-name": "data-hub-match-merge-reader",
+  "description": "Permits reading a matching configuration"
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/hub-internal-config/security/roles/data-hub-match-merge-reader.json]: Is the assumption that this role will allow access to either match or merge configurations? If so, the description should state that.

Comment (srinathgit) [reply marklogic-data-hub/src/main/resources/hub-internal-config/security/roles/data-hub-match-merge-reader.json]: Thanks @rjrudin . I have changed the description.

**Comment by srinathgit** [comment]
Fixed the merge conflict

**Comment by bsrikan** [comment]
There seems to be a test failure.

**Comment by srinathgit** [comment]
> There seems to be a test failure.

Fixed the test failure

**Pull request merged**